worker/worker/worker.js

```diff
@@ -147,50 +147,89 @@ function handleOptions(request) {
     headers.get("Origin") !== null &&
     headers.get("Access-Control-Request-Method") !== null &&
     headers.get("Access-Control-Request-Headers") !== null
-  ){
+  ) {
     // Handle CORS pre-flight request.
     // If you want to check or reject the requested method + headers
     // you can do that here.
     let respHeaders = {
       ...corsHeaders,
-    // Allow all future content Request headers to go back to browser
-    // such as Authorization (Bearer) or X-Client-Name-Version
-      "Access-Control-Allow-Headers": request.headers.get("Access-Control-Request-Headers"),
-    }
+      // Allow all future content Request headers to go back to browser
+      // such as Authorization (Bearer) or X-Client-Name-Version
+      "Access-Control-Allow-Headers": request.headers.get(
+        "Access-Control-Request-Headers"
+      ),
+    };
 
     return new Response(null, {
       headers: respHeaders,
-    })
-  }
-  else {
+    });
+  } else {
     // Handle standard OPTIONS request.
     // If you want to allow other HTTP Methods, you can do that here.
     return new Response(null, {
       headers: {
         Allow: "GET, OPTIONS",
       },
-    })
+    });
   }
 }
 
 async function handler_witness_tweet(request) {
   try {
-    const {searchParams} = new URL(request.url)
-    let pk = decodeURI(searchParams.get('pk'))
-    let handle = searchParams.get('handle')
-    let tweet_id = searchParams.get('tweet_id')
-    const {witness_tweet} = wasm_bindgen;
-    await wasm_bindgen(wasm)
-    const vc = await witness_tweet(TZPROFILES_ME_PRIVATE_KEY, pk, TWITTER_BEARER_TOKEN, handle, tweet_id)
-    return new Response(vc, {status: 200, headers: headers})
+    const { searchParams } = new URL(request.url);
+    let pk = decodeURI(searchParams.get("pk"));
+    let handle = searchParams.get("handle");
+    let tweet_id = searchParams.get("tweet_id");
+    const { witness_tweet } = wasm_bindgen;
+    await wasm_bindgen(wasm);
+    const vc = await witness_tweet(
+      TZPROFILES_ME_PRIVATE_KEY,
+      pk,
+      TWITTER_BEARER_TOKEN,
+      handle,
+      tweet_id
+    );
+    return new Response(vc, { status: 200, headers: headers });
+  } catch (error) {
+    return new Response(error, { status: 500, headers: headers });
+  }
+}
+
+async function handler_discord_message(request) {
+  try {
+    const { witness_discord } = wasm_bindgen;
+    const { searchParams } = new URL(request.url);
+
+    const pk = decodeURI(searchParams.get("pk"));
+    const message_id = searchParams.get("messageId");
+    const channel_id = searchParams.get("channelId");
+    const discord_handle = searchParams.get("discordHandle");
+
+    await wasm_bindgen(wasm); // What is this line for?
```

Comment:
Interfaces with WASM can be a bit messy, at least from what is documented, and CF adds more abstractions. I think it's fair to say that this simply loads the module (and makes the interface actually callable).